test/usage.js

```diff
@@ -313,6 +313,47 @@ describe('usage tests', () => {
     })
   })
 
+  describe('deprecate options', () => {
+    describe('using .option(x, {deprecate: [boolean|string]})', () => {
+      it('{deprecated: true} should show [deprecated]', () => {
+        const r = checkUsage(() => yargs('--help')
+          .option('x', { deprecated: true })
+          .wrap(null)
+          .parse()
+        )
+        r.logs[0].should.include('  -x  [deprecated]')
+      })
+      it('{deprecated: string} should show [deprecated: string]', () => {
+        const r = checkUsage(() => yargs('--help')
+          .option('x', { deprecated: 'string' })
+          .wrap(null)
+          .parse()
+        )
+        r.logs[0].should.include('  -x  [deprecated: string]')
+      })
+    })
+    describe('using .deprecateOption(x, [string])', () => {
```

Comment:
this option should apply `globally` right, could we add a test that a deprecated command set at the top level carries to sub-commands?

Reply:
done